plasma_framework/contracts/src/vaults/EthVault.sol

```diff
@@ -5,29 +5,20 @@ import "./verifiers/IEthDepositVerifier.sol";
 import "../framework/PlasmaFramework.sol";
 
 contract EthVault is Vault {
-    IEthDepositVerifier private _depositVerifier;
-
     event EthWithdrawn(
         address payable indexed target,
         uint256 amount
     );
 
     constructor(PlasmaFramework _framework) Vault(_framework) public {}
 
-    /**
-     * @notice Set the deposit verifier contract. This can be only called by the operator.
-     * @param _contract address of the verifier contract.
-     */
-    function setDepositVerifier(address _contract) public onlyOperator {
-        _depositVerifier = IEthDepositVerifier(_contract);
-    }
-
     /**
      * @notice Allows a user to submit a deposit.
      * @param _depositTx RLP encoded transaction to act as the deposit.
      */
     function deposit(bytes calldata _depositTx) external payable {
-        _depositVerifier.verify(_depositTx, msg.value, msg.sender);
+        swapDepositVerifiersIfNewerGetsEffective();
```

Comment:
As a drawback of chosen approach was mentioned 2-step process, so
1. set new verifier address (annonce it will be changed)
2. swap verifiers if when the quarantine time passed
Then if one wants to deposit (with new tx type) has first check if in-force verifier version match tx.

With current proposition it puts constant cost on every deposit to check condition and first deposit after new verifier maturity time will be penalized with a cost of swap.
Is the cost of these operations significant enough to move to the 2-step process :question: 
WDYT @boolafish @paulperegud

Reply:
Yeah, I think the main issue is the cost of swap. I have a proposal in previous comment, however I don’t know which is better actually haha. (Like two step vs more dirty code) would like to hear your opinion.

Reply:
one thing is to add a small constant cost, but another thing is to force users to pay for the swap